src/AzureDevOpsClient.js

```diff
@@ -0,0 +1,105 @@
+//
+// Copyright (c) Tommy Nguyen
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+//
+
+/**
+ * @typedef {{
+     path: string;
+     line: number;
+     side: "LEFT" | "RIGHT";
+     start_line?: number;
+     start_side?: "LEFT" | "RIGHT";
+     body: string;
+   }} Comment
+ * @typedef {import("azure-devops-node-api/interfaces/common/VsoBaseInterfaces").IRequestOptions} RequestOptions
+ * @typedef {import("azure-devops-node-api/interfaces/GitInterfaces").GitPullRequestCommentThread} GitPullRequestCommentThread
+ */
+
+/**
+ * Establishes a connection to Azure DevOps instance.
+ * @param {string} serverUrl
+ * @param {string} authToken
+ * @param {RequestOptions} [options]
+ */
+function connect(serverUrl, authToken, options) {
+  const azdev = require("azure-devops-node-api");
+  const authHandler = azdev.getPersonalAccessTokenHandler(authToken);
+  const vsts = new azdev.WebApi(serverUrl, authHandler, options);
+  return vsts.connect().then(() => vsts.getGitApi());
+}
+
+/**
+ * Transforms specified comment to a `CommentThread`.
+ * @param {Comment} comment
+ * @param {number} iteration
+ * @returns {GitPullRequestCommentThread}
+ */
+function transformComment({ body, path, line, start_line }, iteration) {
+  const COMMENT_THREAD_STATUS_ACTIVE = 1;
+  const COMMENT_TYPE_TEXT = 1;
+  return {
+    comments: [{ content: body, commentType: COMMENT_TYPE_TEXT }],
+    status: COMMENT_THREAD_STATUS_ACTIVE,
+    threadContext: {
+      filePath: path,
+      rightFileEnd: { line },
+      rightFileStart: start_line ? { line: start_line } : { line },
+    },
+    pullRequestThreadContext: {
+      changeTrackingId: iteration,
+      iterationContext: {
+        firstComparingIteration: iteration,
+        secondComparingIteration: iteration,
+      },
+    },
```

Comment:
What's the significance with of this?

Reply:
It tells ADO which iteration your comment was made on. If we don't provide it, ADO will say that it cannot find the relevant code lines or something to that effect. For most of our use cases, we want to make comments on the very latest iteration.